src/request_executor.cc

```diff
@@ -139,8 +154,15 @@ ResponseRelease(
   switch (memory_type) {
     case TRITONSERVER_MEMORY_CPU:
     case TRITONSERVER_MEMORY_CPU_PINNED: {
-      off_t* offset = reinterpret_cast<off_t*>(buffer_userp);
-      delete offset;
+      AllocationInfo* allocation_info =
+          reinterpret_cast<AllocationInfo*>(buffer_userp);
+      {
+        // Load the data so that it is deallocated automatically.
+        auto result = allocation_info->shm_manager_->Load<char>(
```

Comment:
Found it. Why are we sending unsafe=true?

Reply:
Explained in other comment.